02-express-tutorial/public/main.js

```diff
@@ -0,0 +1,52 @@
+
+//fetching products 
+async function fetchProducts(){
+
+    const url = "http://localhost:3000/api/v1/products";
+
+    try{
+
+        const response = await fetch(url);
+
+        if(!response){
+            const message = `An error has occurred: ${response.status}`
+        }
+
+        const data = await response.json();
+        return data
+
+    }catch(error){
+        console.log(error.message)
+    }
+}
+
+//display products
+async function displayProducts(){
+    let html = '';
+    fetchProducts()
+    .then( response => {
+
+        for( var i=0; i< response.length; i++){
```

Comment:
To answer your question, I think this is totally fine for rendering it to the screen. There are most granular ways to do this but I don't think this is 'wrong' at all. A suggestion I can make is if you want to render all elements in a loop, you can either user the for of loop to make the code more readable and clean. Also that way you can just reference them directly rather than declaring a variable before.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of